### ISSUE TITLE
Flag --memory in minishift start should also accept values in gigabytes

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#1149]
It would be nice to align Minishift's behaviour for --disk and --memory flags. Currently --memory accepts integers meaning megabytes (which is not noted in start's help), but it probably could accept values as: "2048m", "8g", "16g" etc as the --disk flag does.

Current state from help of `start` command:

```
--disk-size string                Disk size to allocate to the Minishift VM. Use the format <size><unit>, where unit = b, k, m or g. (default "20g")
--memory int                      Amount of RAM to allocate to the Minishift VM. (default 2048)
```

**Comment by gbraad** [comment]
As part of #1068 we already `--disk-size` which allows you to specify the size in G/GB. This actually has to deal with inconsistency in how `--memory` is used to assign memory and how to be displayed.

Currently, in `docker-machine`, `minikube` and `minishift`, we assign `2048` as a default for the memory. This will be treated as MB, and results in 2G of memory.

However, in `virt-manager` this VM is created with:

```
$ sudo virsh edit minishift
<memory unit='KiB'>2097152</memory>
```

![image](https://user-images.githubusercontent.com/1894/29348688-24b40b46-8288-11e7-826d-12c604480075.png)

This means that the VM is actually created with more memory assigned: 
#### 2 097 152 KiB = 2.14748365 GB
[Ref](https://www.google.com/search?q=2097152+KiB+in+GB)

This is the reason why the memory assignment was shown as:
```
fmt.Println("   Memory:   ", units.HumanSize(float64((machineConfig.Memory/units.KiB)*units.GB)))
```

The docker-machine `config.json` looks as follows for this VM:
```
{
    "ConfigVersion": 3,
    "Driver": {
        "Memory": 2048,
        "DiskSize": 20000,
```

**Comment by gbraad** [comment]
in the KVM driver plugin actually `M` is being used:

https://github.com/dhiltgen/docker-machine-kvm/blob/master/kvm.go#L37
https://github.com/dhiltgen/docker-machine-kvm/blob/master/kvm.go#L104

**Comment by gbraad** [comment]
Fixed related issue in `docker-machine-driver-kvm`: https://github.com/dhiltgen/docker-machine-kvm/issues/54#issuecomment-322721222

**Comment by hferentschik** [comment]
Merged via pull request #1283
